streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentInfo.java

```diff
@@ -129,19 +122,33 @@ public ByteBuffer encode() {
         try (final DataOutputStream out = new DataOutputStream(baos)) {
             switch (usedVersion) {
                 case 1:
-                    encodeVersionOne(out);
+                    out.writeInt(usedVersion); // version
+                    encodeActiveAndStandbyTaskAssignment(out);
                     break;
                 case 2:
-                    encodeVersionTwo(out);
+                    out.writeInt(usedVersion); // version
+                    encodeActiveAndStandbyTaskAssignment(out);
+                    encodePartitionsByHost(out);
                     break;
                 case 3:
-                    encodeVersionThree(out);
+                    out.writeInt(usedVersion);
+                    out.writeInt(commonlySupportedVersion);
+                    encodeActiveAndStandbyTaskAssignment(out);
+                    encodePartitionsByHost(out);
                     break;
                 case 4:
-                    encodeVersionFour(out);
+                    out.writeInt(usedVersion);
+                    out.writeInt(commonlySupportedVersion);
+                    encodeActiveAndStandbyTaskAssignment(out);
+                    encodePartitionsByHost(out);
+                    out.writeInt(errCode);
                     break;
                 case 5:
-                    encodeVersionFive(out);
+                    out.writeInt(usedVersion);
+                    out.writeInt(commonlySupportedVersion);
+                    encodeActiveAndStandbyTaskAssignment(out);
+                    encodePartitionsByHostAsDictionary(out);
+                    out.writeInt(errCode);
```

Comment:
Having to bounce through an extra method for each version does nothing but make the protocol harder to understand. After inlining, we can clearly see the structure of each protocol version, and compare it with the decode method for symmetry.

I've also inlined other encode/decode helper methods, for exactly the same reason. I won't comment on other instances.

Reply:
This looks so much better!

prop: drop inconsistently used version comments.

Reply:
Ah, didn't notice that. Probably an artifact of the inlining.

It _is_ annoying, but if there are no blockers, I might defer this to the future, so we don't have to wait for another passing build.